pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -72,7 +75,12 @@ func ApplyLocalityLBSetting(
 		// Failover needs outlier detection, otherwise Envoy will never drop down to a lower priority.
 		// Do not apply default failover when locality LB is disabled.
 	} else if enableFailover && (localityLB.Enabled == nil || localityLB.Enabled.Value) {
-		applyLocalityFailover(locality, loadAssignment, localityLB.GetFailover())
+		if len(localityLB.FailoverPriority) > 0 {
+			applyPriorityFailover(loadAssignment, wrappedLocalityLbEndpoints, proxyLabels, localityLB.FailoverPriority)
+			return
+		}
+		// TODO: apply failover only when it is explicitly specified
```

Comment:
Why?

Reply:
That seems like a breaking API change

Reply:
The previous behavior seems not quite right to me. If we have set an empty localityLoadbalancingSetting: {}, we apply failover. I cannot remember that.

Reply:
Yes, `failover` is just to further customize the failover settings.

I agree the API is bad but we cannot change it

Reply:
OK, i will remove the `TODO`